File: 0020-valid-parentheses/2T1/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let stack = [];
    const brackets = {
        "[" : "]",
        "(" : ")",
        "{" : "}"
    }
    for(const i of s){
        if('([{'.includes(i)) stack.push(i);
        else if(brackets[stack.pop()] != i) return false;
    }
    return stack.length === 0;
};

File: 0020-valid-parentheses/Lwin/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        int i;
        char[] in = s.toCharArray();
        Stack<Character> out = new Stack<>();
        
        for (i=0; i<in.length; i++) {
            if (in[i] == '(') {
                out.push(')');
            }
            else if (in[i] == '[') {
                out.push(']');
            }
            else if (in[i] == '{') {
                out.push('}');
            }
            else if (in[i] == ')' || in[i] == ']' || in[i] == '}') {
                if (out.empty() || out.pop() != in[i]) {
                    return false;
                }
            }
        }
        return out.empty();
    }
}